src/ray/raylet/node_manager.cc

```diff
@@ -479,12 +519,12 @@ void NodeManager::Heartbeat() {
     // Add resource load by shape. This will be used by the new autoscaler.
     auto resource_load = local_queues_.GetResourceLoadByShape(
         RayConfig::instance().max_resource_shapes_per_load_report());
-    heartbeat_data->mutable_resource_load_by_shape()->Swap(&resource_load);
+    resources_data->mutable_resource_load_by_shape()->Swap(&resource_load);
   }
 
   // Set the global gc bit on the outgoing heartbeat message.
   if (should_global_gc_) {
-    heartbeat_data->set_should_global_gc(true);
+    resources_data->set_should_global_gc(true);
```

Comment:
We should probably move this to the heartbeat? Thoughts?

Reply:
I'd like keeping heartbeat as simple as it can. Better not be coupled with node manager or resources related.
@raulchen also points out that `should_global_gc` is not very proper(even acceptable) to be in resources data, and maybe it's better to use a individual rpc message.